src/main/java/Duke.java

```diff
@@ -1,10 +1,331 @@
+                    for(int i=0; i < allTasks.size(); i++) {
+                        if (allTasks.get(i) != null) {
+                            taskNum = i + 1;
+                            System.out.println(taskNum + ". " + allTasks.get(i).toString());
+                        }
+                    }
+                    System.out.println("----------------------------------------------------");
+                    break;
+                case "todo":
+                    try {
+                        if (actionArr.length == 1) {
+                            throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+                        }
+                    } catch (DukeException b){
+                        System.out.println(b);
+                        break;
+                    }
+                    Todo t = new Todo(actionArr[1]);
+                    allTasks.add(t);
+                    System.out.println("----------------------------------------------------");
+                    System.out.println("Got it. I've added this task:");
+                    t.toString();
+                    System.out.println("Now you have " + allTasks.size() + " tasks in the list.");
+                    System.out.println("----------------------------------------------------");
+                    taskNum = allTasks.size();
+                    if (taskNum==1) {
+                        try {
+                            writeToFile(filePath, taskNum + ". " + allTasks.get(taskNum-1).toString() + "\n");
+                        } catch (IOException e) {
+                            System.out.println("Something went wrong: " + e.getMessage());
+                        }
+                    } else {
+                        try {
+                            appendToFile(filePath, taskNum + ". " + allTasks.get(taskNum-1).toString() + "\n");
+                        } catch (IOException e) {
+                            System.out.println("Something went wrong: " + e.getMessage());
+                        }
+                    }
+                    break;
+                case "deadline":
+                    try {
+                        if (actionArr.length == 1) {
+                            throw new DukeException("OOPS!!! Deadline description cannot be empty.");
+                        }
+                    } catch (DukeException c){
+                        System.out.println(c);
+                        break;
+                    }
+                    String[] deadlineParts = actionArr[1].split("/by");
+                    try {
+                        if (deadlineParts.length == 1) {
+                            throw new DukeException("OOPS!! You did not include a deadline using /by. :(");
+                        }
+                    } catch (DukeException d){
+                        System.out.println(d);
+                        break;
+                    }
```

Comment:
You can consider refactoring this code (the entire switch statement) by implementing SLAP. This is a big chunk of code and reduces readability. See [SLAP](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) and [avoid long methods](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)